### ISSUE TITLE
jenkins.war is not in the classpath error when trying to run tests

### ISSUE THREAD

**Issue by jakub-bochenski** [jenkinsci/gitlab-plugin#671]
I can't get this project to run tests locally.

I'm always getting:
```
GitLabEnvironmentContributorTest.com.dabsquared.gitlabjenkins.environment.GitLabEnvironmentContributorTest jenkins.war is not in the classpath. If you are running this from the core workspace, run 'mvn install' to create the war image in war/target/jenkins
 ```

I tried running:  `mvn clean install`, `mvn clean install -DskipTests` but the result is the same.
I don't see anything under `war/target/jenkins` after running those.

I tried updating the hpi plugin to latest (2.1) but it doesn't help.

I tried manually extracting `jenkins-war-1.609.3-war-for-test.jar` into `war/target/jenkins` but it doesn't help

```
$ mvn --version 
Apache Maven 3.3.9
Java version: 1.8.0_131, vendor: Oracle Corporation
OS name: "linux", version: "4.4.0-97-generic", arch: "amd64", family: "unix"
```

**Comment by omehegan** [comment]
@mreichel or @Argelbargel any idea why this might be happening? It's not an issue for me, but I am far from a maven guru.

**Comment by Argelbargel** [comment]
Sorry, i can not reproduce this. I've had a similar problem when building a gradle-based plugin-project against jenkins version > 2.63 (see [JENKINS-46899](https://issues.jenkins-ci.org/browse/JENKINS-46899)). But this seems to have a different cause as we're using version 1.6x to build.

**Comment by mreichel** [comment]
I can't reproduce it either. @jakub-bochenski, do you have any more information? Do you run maven from an IDE (e.g. Intellij has a bug there, updating the parent in the pom should fix it)?

**Comment by jakub-bochenski** [comment]
@mreichel I tried this from CLI

Maybe we could try to debug this another way. 
Can you point me to how/when exactly the jenkins.war image is supposed to be put in `war/target/jenkins`?
Then I could try and check why isn't it happening on my system.

**Comment by mreichel** [comment]
Well, we inherit the functionality from the hpi-plugin and the Jenkins test harness (that should be executed in the test-compile phase). I think it was fixed in [Jenkins test harness](https://github.com/jenkinsci/jenkins-test-harness/compare/2a12efb35ad1...b45ae9bb0299). That should be included in a parent version > 2.35. You could also try to skip test compilation with
`mvn install -Dmaven.test.skip=true`

**Comment by omehegan** [comment]
Seems to be an env issue, not a plugin issue. Also works fine for me with maven 3.5.2.
